pkg/scheduler/queue/block_queue.go

```diff
@@ -380,15 +380,15 @@ func (p *BlockQueue) AddUnschedulableIfNotPresent(unitInfo *framework.QueuedUnit
 	queue, prevUnitInfo, exists := p.findAndRemoveUnitFromQueue(unitInfo.UnitKey)
 	if exists {
 		klog.V(4).InfoS("SchedulingQueue AddUnschedulable, exist same unit and will merge them", "unitKey", unitInfo.UnitKey, "oldQueue", queue)
-		// We add the unitInfo to prevUnitInfo cause we may update the old pods.
-		prevUnitInfo.AddPods(unitInfo.GetPods())
-		unitInfo = prevUnitInfo
+		// We merge the unitInfo into prevUnitInfo cause we may update the old pods.
```

Comment:
this function will not update existing (old) pods ?

Reply:
yes, `unitInfo` has the latest version of existing pods, so should not update existing pods, only add new coming pods.

Reply:
I will update the comment later.